src/js/form-model.js

```diff
@@ -359,6 +359,19 @@ FormModel.prototype.mergeXml = function( recordStr ) {
     recordStr = recordStr.replace( /<!--[^>]*-->/g, '' );
     record = parser.parseFromString( recordStr, 'text/xml' );
 
+    /**
+     * Insert missing relevant nodes
+     */
+    //get the parent element of the relevant node
+    //get node that comes before this relevant node in template elements
+    // get all the parent elements of the nodes in the record
+    // var relevantParents = record.querySelectorAll("repeat111");
+    //check if one of it's children is the relevant node, and if not then
+    // insert the relevant node after the node above for each parent element
+
+    // repeat for all relevant node
+    // - we need to first find all relevant nodes
+
```

Comment:
No, this would not be the right approach. I suspect the issue is with merging the record into the model, so that's where the fix would be.